pycbc/inference/burn_in.py

```diff
@@ -415,13 +476,23 @@ def nacl(self, filename):
         """Carries out the nacl test and stores the results."""
         pass
 
+    def chain_moments(self):
+        """Carries out the chain_moments test and stores the results. """
+        pass
+
```

Comment:
For the base class make this an abstractmethod since you expect it to be implemented for all types of MCMC samplers.

Since you haven't implemented a version that works for parallel tempered samplers yet, add a `chain_moments` method to the `*MultiTempered*` class that just raises a `NotImplementedError`.